// @progress/kendo-listbox-vue-wrapper/dist/npm/KendoListBox/KendoListBoxMixin.js
'use strict';

Object.defineProperty(exports, "__esModule", {
    value: true
});

var _kendoBaseComponentsVueWrapper = require('@progress/kendo-base-components-vue-wrapper');

var _ListBoxProps = require('./ListBoxProps');

var _ListBoxProps2 = _interopRequireDefault(_ListBoxProps);

var _Draggable = require('./Draggable');

var _Draggable2 = _interopRequireDefault(_Draggable);

var _Messages = require('./Messages');

var _Messages2 = _interopRequireDefault(_Messages);

var _Toolbar = require('./Toolbar');

var _Toolbar2 = _interopRequireDefault(_Toolbar);

function _interopRequireDefault(obj) { return obj && obj.__esModule ? obj : { default: obj }; }

exports.default = {
    template: '<select><slot></slot></select>',
    props: _ListBoxProps2.default,
    mixins: [_kendoBaseComponentsVueWrapper.KendoBaseComponent, _kendoBaseComponentsVueWrapper.KendoBaseDatasourceComponent, _Draggable2.default, _Messages2.default, _Toolbar2.default],
    mounted() {
        kendo.jQuery(this.$el).kendoListBox(this.widgetOptions);

        this.$emit('kendowidgetready', this.kendoWidget());
    },
    methods: {
        kendoWidget() {
            return kendo.jQuery(this.$el).getKendoListBox();
        }
    }
};
